community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/NativeLabelScanStore.java

```diff
@@ -342,6 +378,50 @@ public void recoveryCompleted( long numberOfPagesVisited, long numberOfCleanedCr
         };
     }
 
+    private static final int HEADER_LENGTH = 1;
+    private static final byte CLEAN = (byte) 0x00;
+    private static final byte DIRTY = (byte) 0x01;
+
+    private class GetDirtyReader implements Header.Reader
+    {
+        private boolean isDirty;
+        private boolean read;
+
+        @Override
+        public void read( PageCursor from, int length )
+        {
+            byte[] header = new byte[HEADER_LENGTH];
+            from.getBytes( header );
+            read = true;
+            isDirty = header[0] == DIRTY;
```

Comment:
Why not just `from.getByte()` and skip the array construction and `HEADER_LENGTH`?